x-pack/plugins/security_solution/public/entity_analytics/components/asset_criticality_file_uploader/validations.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  CRITICALITY_CSV_MAX_SIZE_BYTES,
+  CRITICALITY_CSV_MAX_SIZE_BYTES_WITH_TOLERANCE,
+  parseAssetCriticalityCsvRow,
+} from '../../../../common/entity_analytics/asset_criticality';
+import { SUPPORTED_FILE_EXTENSIONS, SUPPORTED_FILE_TYPES } from './constants';
+
+export interface RowValidationErrors {
+  error: string;
+  index: number;
+}
+
+export const validateParsedContent = (
+  data: string[][]
+): { valid: string[][]; invalid: string[][]; errors: RowValidationErrors[] } => {
+  if (data.length === 0) {
+    return { valid: [], invalid: [], errors: [] };
+  }
+
+  let errorIndex = 1; // Error index starts from 1 because EuiCodeBlock line numbers start from 1
```

Comment:
seems like this index is simply `acc.errors.length + 1`? If so, I'd rather remove the variable increment as that'd be less code that could go wrong

Reply:
This index is a hack because `EuiCodeBlock` doesn't allow showing consecutive line numbers.
I use it inside each error `acc.errors.push({ error: parsedRow.error, index: errorIndex });` to know in which line they will be placed.
We could also get the index from the map function, and +1, as you suggested. But it might be unclear 🤔 
I will try it out.

Reply:
I aborted this change to keep all errors with the same type. Otherwise, some errors will be indexed by the array positioned others by the `index` field.